File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        
        int count = 0;
        for(int i = 0; i < s.length(); i++)
        {
            count += isPalindrome(s, i, i);
            count += isPalindrome(s, i, i + 1);
        }
        
        return count;
        
    }
    
    public int isPalindrome(String s, int left, int right)
    {
        int count = 0;
        while(left >= 0 && right <= s.length() - 1)
        {
            if(s.charAt(left) != s.charAt(right))
                return count;
            
            ++count;
            --left;
            ++right;
        }
        
        return count;
    }
}

// a a b a a
// a, a, b, a, a, aa, aba, aa, aabaa

File: Medium/0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution(object):   
    def countSubstrings(self, s):
        count = 0
        for c in range(len(s)):            
            count += self.isPalinedrome(s, c, c)                               
            count += self.isPalinedrome(s, c, c + 1)

        return count
    
    def isPalinedrome(self, s, left, right):
        count = 0
        while left >= 0 and right <= len(s) - 1:
            if s[left] != s[right]:
                return count;
            
            count += 1
            left -= 1
            right += 1
            
        return count;
    
    
    
            
        
        